### ISSUE TITLE
Address descriptions of the dynamic size of the arrows to handle large edge case

### ISSUE THREAD

**Issue by zepumph** [phetsims/ohms-law#97]
From https://github.com/phetsims/ohms-law/issues/90#issuecomment-338188950. Since the largest size of the arrows goes off the screen, we should instead use a height of arrows that is smaller but still "huge." Then make anything bigger than that also huge.

Another suggestion is to have an exponential relationship rather than a linear one.

**Comment by zepumph** [comment]
As part of https://github.com/phetsims/ohms-law/issues/90 @terracoda said:

>@zepumph, it looks like the description for the Current Arrows changes more often, but the size description still doesn't match what I think it should be visually.

>@zepumph re my last comment, would it be possible to map out the 7 sizes visually, and see how that fits with the model. It's possible that the ranges for each size are not even.

I'll look into it and see what is actually happening in the model. It's possible that they currently aren't even, or it is possible that they are based on something that we don't expect (like a model property rather than the actual size of the arrows. I'll get back to you @terracoda.

**Comment by terracoda** [comment]
@zepumph, we will  have similar challenges with describing the properties of the wire in RIAW as well, so what we learn here will be used again immediately.

**Comment by zepumph** [comment]
I suspect that this is a similar problem with the size of 'I' (current), but I'm going to focus on the arrows right now.

**Comment by zepumph** [comment]
I first looked at a strategy that involed making a piece-wise linear function mapping the first half of the values linearly up to the default, and then the second half linearly up to the max. That didn't work because the default isn't really at our halfway point, so then I updated it so that the "default" was more at our 3/4s mark, and then you map linearly on both sides based on the number of indices, like so:

    getArrowSizeDescription: function() {

      var range = this.arrowHeightRange;
      var height = this.bottomLeftArrow.height;

      var index = 0;
      var threeQuartersIndex = Math.floor( 3 * OhmsLawConstants.RELATIVE_SIZE_STRINGS.length / 4 );
      if ( height <= range.getDefaultValue() ) {
        index = Util.roundSymmetric( Util.linear( range.min, range.getDefaultValue(), 0, threeQuartersIndex, height ) );
      }
      else {
        index = Util.roundSymmetric( Util.linear( range.getDefaultValue(), range.max, threeQuartersIndex, OhmsLawConstants.RELATIVE_SIZE_STRINGS.length - 1, height ) );
      }
```
The goal of the above algorithm would be to make a smaller slope for the smaller sizes, and a larger slope for the larger values. This didn't really have the control that I was hoping for.

**Comment by zepumph** [comment]
Next I went with a capping strategy, basically I said that the linear mapping goes from the smallest arrow height to a fixed point, and everything larger than that is just huge. The fixed point is based on the height of the wire box, so I think it is pretty robust. I also feel like there is good control throughout the middle ranges. @terracoda @emily-phet please review here: https://www.colorado.edu/physics/phet/dev/html/ohms-law/1.4.0-dev.15/ohms-law-a11y-view.html

**Comment by jessegreenberg** [comment]
@zepumph it feels really nice to me!

**Comment by zepumph** [comment]
It seems like we won't have to do this for the I in the formula node, since it is based on a comparison of V, we may though find that minor tweakings should be done for the comparison ranges, that is a design decision though.

**Comment by terracoda** [comment]
@zepumph, it is looking good. Regarding your comment 
> I said that the linear mapping goes from the smallest arrow height to a fixed point, and everything larger than that is just huge.

I did something similar, though I used the size of letter I as my guide rather than the bounds of the wire box. I think both are robust. In my design thinking, the arrows are "very large" when letter I is at or just inside the bounds of the top of the sim (the largest size for V and R), and the arrows are  "huge" when letter I goes beyond the bounds of the sim.

Would that slight change to the "huge" fixed-point threshold work?

**Comment by zepumph** [comment]
Because of the way it is implemented, it would be difficult to implement this sizing based exatly on the intersection of the 'I' and the screen bounds, so I eyeballed it. I don't think it is too sloppy though, since they are both based on constants and I documented the correlation in the code. @terracoda please take a look on phettest.

**Comment by terracoda** [comment]
@zepumph I think you did a good eyeball estimate. The descriptions look/sound better to me know.
We might have to do something similar in RIAW.
I think we can close this issue.
